src/tls/rustls.rs

```diff
@@ -121,13 +124,32 @@ where
 }
 
 #[async_trait::async_trait]
-impl<S> Reconnect for TlsStream<S>
-where
-    S: AsyncRead + AsyncWrite + Send + Unpin + Reconnect,
-{
-    async fn reconnect(&mut self) -> io::Result<Self> {
-        let stream = self.stream.get_mut().0.reconnect().await?;
-        TlsStream::new(stream, self.connector.clone(), self.hostname.clone()).await
+impl Reconnect for BufStream<TlsStream<tokio::net::TcpStream>> {
```

Comment:
Can we have a blanket:
```rust
impl<S> Reconnect for BufStream<S> where S: Reconnect
```

and then just `impl Reconnect for TlsStream<TcpStream>`?

Reply:
In fact, maybe we should _also_ have:
```rust
impl<S> Reconnect for TlsStream<S> where S: Reconnect
```
rather than specializing this to `TlsStream<TcpStream>`?

Reply:
This would mean for `native_tls` we also only need a single impl and that's

```rust
impl<S> Reconnect for TlsStream<S> where S: Reconnect
```

(with `native_tls::TlsStream`)

Reply:
please see [comment](https://github.com/jonhoo/faktory-rs/pull/64#discussion_r1702745387)